### PR TITLE
feat(ci): Add workflow for creating new release tag

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR adds a new workflow for creating releases

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Update the VERSION
+        run: |
+          echo "$VERSION" > VERSION
+        env:
+          VERSION: ${{ github.event.inputs.tag }}
+
+      - name: Commit changes and tag it
+        run: |
+          git add VERSION
+          git commit -m "ci: update VERSION to $VERSION"
```
Comment (SamYuan1990) [inline .github/workflows/release.yaml:38]: will this commit -m breaks DCO check?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: "v${{ github.event.inputs.tag }}"
+          release_name: "v${{github.event.inputs.tag}}"
+          draft: false
+          prerelease: false
```
Comment (SamYuan1990) [inline .github/workflows/release.yaml:57]: add a change log file?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Git set user
+        run: |
+          git config  user.name "$USERNAME"
+          git config  user.email "$USERNAME-[bot]@users.noreply.github.com"
+        env:
+          USERNAME: ${{ github.actor }}
+
+      - name: Update the VERSION
+        run: |
+          echo "$VERSION" > VERSION
```
Comment (SamYuan1990) [inline .github/workflows/release.yaml:31]: I am not sure, if we can use shell script to do a math function. for example read 0.0.3 from current file auto increases to 0.0.4 and commit?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (vprashar2929) [reply .github/workflows/release.yaml:38]: No it won't

File: .github/workflows/release.yaml (hunk 3 above)
Comment (vprashar2929) [reply .github/workflows/release.yaml:31]: hmm, we can do that as well. But I wanted the version to be provided during runtime in cases where we want to break the continuous flow and maybe release something like `0.1.0`

File: .github/workflows/release.yaml (hunk 2 above)
Comment (vprashar2929) [reply .github/workflows/release.yaml:57]: Will add it in the next PR

**Review by SamYuan1990 — APPROVED** [review]

**Pull request merged**